DB/api/db.py

```diff
@@ -66,6 +75,80 @@ def query(parsed_url):
 		return format_data(header, results, json)
 
 
+def generate_total_energy(device_id, start_time, end_time, channel):
+	"""
+	Returns total energy for a particular "channel" (device in table data_raw),
+	over a specified time period
+
+	:param device_id: The serial number of the device in question
+	:param start_time: The start of the time range for which to query for data
+	:param end_time: The end of the time range for which to query for data
+	:param channel: channel filter
+	"""
+
+	# Initialize parameter list and WHERE clause
+	params = [device_id]
+	query = "SELECT CAST(data as DECIMAL) / 36e5 FROM " + TABLE + " as raw WHERE serial = %s AND type = 'P'"
+
+	# Generate WHERE clause
+	if start_time and end_time and channel:
+		query += " AND device = %s AND (time = to_timestamp(%s) OR time = to_timestamp(%s));"
+		params.append(channel)
+		params.append(start_time)
+		params.append(end_time)
+		rows = perform_query(query, tuple(params))
+	else:
+		raise Exception("start_time, end_time, or device not provided")
+
+	if len(rows) > 1:
+		total_energy = abs(abs(rows[0][0]) - abs(rows[1][0]))
+	else:
+		raise Exception("No power data for the timestamps specified")
+
+	result = '{ total_energy: ' + str(total_energy) + '}'
+	return result
+
+
+def generate_energy_list(device_id, start_time, end_time, channel, granularity):
```

Comment:
Why is the function needed? We already have diffing support. Just add chunking to the existing query builder.